pkg/integrations/ai_agent/tool_call_manager.go

```diff
@@ -604,3 +596,199 @@ func (m *DefaultToolCallManager) buildOutputItemsFromIntegrationOutput(output do
+
+		// Update the resolved value if it changed
+		if resolvedValue != fieldValue {
+			resolvedParams.ResolvedSettings[fieldKey] = resolvedValue
+
+			// Update resolution log
+			for i, res := range resolvedParams.ResolutionLog {
+				if res.Path == fieldKey {
+					resolvedParams.ResolutionLog[i].Value = resolvedValue
+					break
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// extractDependentValues extracts values that a peekable field depends on
+func (m *DefaultToolCallManager) extractDependentValues(fieldKey string, resolvedSettings map[string]interface{}, toolDef *ToolDefinition) map[string]interface{} {
+	dependentValues := make(map[string]interface{})
+
+	// Get integration schema to find dependent properties
+	integration, err := m.executorIntegrationManager.GetIntegration(context.Background(), toolDef.ToolExecutor.IntegrationType)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("integration_type", string(toolDef.ToolExecutor.IntegrationType)).
+			Msg("Failed to get integration for dependent value extraction")
+		return dependentValues
+	}
+
+	action := m.findActionByType(integration.Actions, toolDef.ActionType)
+	if action == nil {
+		return dependentValues
+	}
+
+	prop := m.findPropertyByKey(action.Properties, fieldKey)
+	if prop == nil || len(prop.PeekableDependentProperties) == 0 {
+		return dependentValues
+	}
+
+	for _, dep := range prop.PeekableDependentProperties {
+		if value, exists := resolvedSettings[dep.PropertyKey]; exists {
+			dependentValues[dep.PropertyKey] = value
+		}
+	}
+
+	return dependentValues
+}
+
+// performPeekableResolution performs the actual peekable resolution using the tool executor
+func (m *DefaultToolCallManager) performPeekableResolution(ctx context.Context, displayValue interface{}, peekableType domain.IntegrationPeekableType, toolDef *ToolDefinition) (interface{}, error) {
+	// Convert display value to string for processing
+	displayStr := fmt.Sprintf("%v", displayValue)
+	if displayStr == "" || displayStr == "<nil>" {
+		return displayValue, nil // Return original if empty
+	}
+
+	// Check if the tool executor supports peeking
+	peeker, ok := toolDef.ToolExecutor.Executor.(domain.IntegrationPeeker)
+	if !ok {
+		return displayValue, nil
+	}
+
+	// Prepare peek parameters
+	peekParams := domain.PeekParams{
+		PeekableType: peekableType,
+		WorkspaceID:  toolDef.ToolExecutor.WorkspaceID,
+		UserID:       "", // Will be set by the integration if needed
+		PayloadJSON:  []byte("{}"), // Default empty payload
+	}
+
+	// Perform the peek operation
+	peekResult, err := peeker.Peek(ctx, peekParams)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("peekable_type", string(peekableType)).
+			Msg("Peekable API call failed")
+		return displayValue, fmt.Errorf("peekable lookup failed: %w", err)
+	}
+
+	// Debug: Log what peekable API returned
+	// Process peekable results
+
+	// Search for matching value in peek results
+	resolvedValue := m.findMatchingPeekValue(displayStr, peekResult.Result)
```

Comment:
You are currently passing all peekable list to tool call arguments, but i think we should not do it, instead of it we should provide peekable data in thinking stage.